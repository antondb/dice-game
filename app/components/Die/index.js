/**
 *
 * Die
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const StyledDie = styled(({ isRolling, ...rest }) => (
  <FontAwesomeIcon {...rest} />
))`
  color: #f5f0f7;
  font-size: 2em;

  animation: ${({ isRolling }) =>
    isRolling ? ' spin 1s linear infinite' : 'none'};

  @keyframes spin {
    100% {
      transform: rotate(360deg);
    }
  }
`;

const StyledDiceCup = styled.div`
  border: solid 0.5em #968399;
  padding: 1em 0em;
  height: 5em;
  text-align: center;
  width: 5em;
  display: inline-block;
  border-radius: 50%;
`;

const numberMap = new Map();
numberMap.set(1, 'one');
numberMap.set(2, 'two');
numberMap.set(3, 'three');
numberMap.set(4, 'four');
numberMap.set(5, 'five');
numberMap.set(6, 'six');

function Die(props) {
  const { value, isRolling } = props;

  const dieIcon = value === null ? 'star' : `dice-${numberMap.get(value)}`;
  return (
    <StyledDiceCup>
      <StyledDie isRolling={isRolling} icon={dieIcon} />
    </StyledDiceCup>
  );
}

Die.propTypes = {
  /** The Value of the die */
  value: PropTypes.number,
  /** Should the rolling animation be showing */
  isRolling: PropTypes.bool,
};

export default Die;
