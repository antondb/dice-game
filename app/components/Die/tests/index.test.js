/**
 *
 * Tests for Die
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import { library } from '@fortawesome/fontawesome-svg-core';
import {
  faDiceOne,
  faDiceTwo,
  faDiceThree,
  faDiceFour,
  faDiceFive,
  faDiceSix,
  faStar,
} from '@fortawesome/free-solid-svg-icons';

import Die from '../index';

library.add(
  faDiceOne,
  faDiceTwo,
  faDiceThree,
  faDiceFour,
  faDiceFive,
  faDiceSix,
  faStar,
);

describe('<Die />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Die isRolling={false} value={2} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Die isRolling={false} value={2} />);
    expect(firstChild).toMatchSnapshot();
  });
});
