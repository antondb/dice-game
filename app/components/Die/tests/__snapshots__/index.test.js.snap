// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Die /> Should render and match the snapshot 1`] = `
<div
  class="Die__StyledDiceCup-sc-5aaqsf-1 bNWieR"
>
  <svg
    aria-hidden="true"
    class="svg-inline--fa fa-dice-two fa-w-14 Die__StyledDie-sc-5aaqsf-0 jaCaeW"
    data-icon="dice-two"
    data-prefix="fas"
    focusable="false"
    role="img"
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M384 32H64C28.65 32 0 60.65 0 96v320c0 35.35 28.65 64 64 64h320c35.35 0 64-28.65 64-64V96c0-35.35-28.65-64-64-64zM128 192c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32zm192 192c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
      fill="currentColor"
    />
  </svg>
</div>
`;
